src/libostree/ostree-fetcher-curl.c

```diff
@@ -575,6 +591,178 @@ write_cb (void *ptr, size_t size, size_t nmemb, void *data)
+
+  if (len != 29)
+    return NULL;
+
+  const char *day_name = buf;
+  const char *day = buf + 5;
+  const char *month_name = day + 3;
+  const char *year = month_name + 4;
+  const char *hour = year + 5;
+  const char *minute = hour + 3;
+  const char *second = minute + 3;
+  const char *tz = second + 3;
+
+  for (day_name_index = 0; day_name_index < G_N_ELEMENTS (day_names); day_name_index++)
+    {
+      if (strncmp (day_names[day_name_index], day_name, 3) == 0)
+        break;
+    }
+  if (day_name_index >= G_N_ELEMENTS (day_names))
+    return NULL;
+  /* don’t validate whether the day_name matches the rest of the date */
+  if (*(day_name + 3) != ',' || *(day_name + 4) != ' ')
+    return NULL;
+  if (!parse_uint (day, 2, 1, 31, &day_int))
+    return NULL;
+  if (*(day + 2) != ' ')
+    return NULL;
+  for (month_name_index = 0; month_name_index < G_N_ELEMENTS (month_names); month_name_index++)
+    {
+      if (strncmp (month_names[month_name_index], month_name, 3) == 0)
+        break;
+    }
+  if (month_name_index >= G_N_ELEMENTS (month_names))
+    return NULL;
+  if (*(month_name + 3) != ' ')
+    return NULL;
+  if (!parse_uint (year, 4, 0, 9999, &year_int))
+    return NULL;
+  if (*(year + 4) != ' ')
+    return NULL;
+  if (!parse_uint (hour, 2, 0, 23, &hour_int))
+    return NULL;
+  if (*(hour + 2) != ':')
+    return NULL;
+  if (!parse_uint (minute, 2, 0, 59, &minute_int))
+    return NULL;
+  if (*(minute + 2) != ':')
+    return NULL;
+  if (!parse_uint (second, 2, 0, 60, &second_int))  /* allow leap seconds */
+    return NULL;
+  if (*(second + 2) != ' ')
+    return NULL;
+  if (strncmp (tz, "GMT", 3) != 0)
+    return NULL;
+
+  return g_date_time_new_utc (year_int, month_name_index + 1, day_int,
+                              hour_int, minute_int, second_int);
+}
+
+/* CURLOPT_HEADERFUNCTION */
+static size_t
+response_header_cb (const char *buffer, size_t size, size_t n_items, void *user_data)
+{
+  const size_t real_size = size * n_items;
+  GTask *task = G_TASK (user_data);
+  FetcherRequest *req;
+
+  req = g_task_get_task_data (task);
+
+  if (real_size > strlen ("ETag: ") &&
+      strncasecmp (buffer, "ETag: ", strlen ("ETag: ")) == 0)
+    {
+      g_clear_pointer (&req->out_etag, g_free);
+      req->out_etag = g_strstrip (g_strdup (buffer + strlen ("ETag: ")));
+    }
+  else if (real_size > strlen ("Last-Modified: ") &&
+           strncasecmp (buffer, "Last-Modified: ", strlen ("Last-Modified: ")) == 0)
+    {
+      g_autoptr(GDateTime) dt = NULL;
+      const char *lm_buf = buffer + strlen ("Last-Modified: ");
+      size_t lm_len = real_size - strlen ("Last-Modified: ");
+
+      /* Check termination */
+      if (lm_len < 2 ||
+          lm_buf[lm_len - 2] != '\r' ||
+          lm_buf[lm_len - 1] != '\n')
+        dt = NULL;
+      else
+        dt = parse_rfc2616_date_time (lm_buf, lm_len - 2);
+
+      req->out_last_modified = (dt != NULL) ? g_date_time_to_unix (dt) : 0;
```

Comment:
Do we need an `if (req->out_last_modified) g_date_time_unref (req->out_last_modified);` here?

Reply:
No, `req->out_last_modified` is a `guint64`